src/test/kotlin/dk/cachet/carp/webservices/data/service/CoreDataStreamServiceTest.kt

```diff
@@ -0,0 +1,133 @@
+import java.io.File
+import kotlin.test.assertFailsWith
+@OptIn(ExperimentalCoroutinesApi::class)
+class CoreDataStreamServiceTest {
+    private val dataStreamSequenceRepository: DataStreamSequenceRepository = mockk()
+    private val objectMapper: ObjectMapper = mockk()
+    private val json = createDefaultJSON(COMMON_SERIAL_MODULE)
+
+    companion object
+    {
+        private val dataStreamConfigurationRepository: DataStreamConfigurationRepository = mockk()
+
+        /** Data paths */
+
+        const val DATA_STREAM_SERVICE_PATH = "/data-stream-service.json"
+    }
+
+    @Nested
+    inner class AppendToDataStreams {
+
+        private val mutableDataStreamBatch: MutableDataStreamBatch = mockk()
+
+        @Test
+        fun `test appends batch to mutableDataStreamBatch` () = runTest {
+            val objectMapper = mockk<ObjectMapper>()
+            val sut = CoreDataStreamService(
+                mockk(),
+                mockk(),
+                mockk(),
+                objectMapper
+            )
+
+            every { dataStreamSequenceRepository.saveAll(any<Iterable<DataStreamSequence>>()) } returns mockk()
+            every { mutableDataStreamBatch.appendBatch(any()) } answers {
+                println("Append batch called with argument: ${args[0]}")
+                mockk() // Return a mock DataStreamBatch
+            }
+
+            val batch = readBatchFromJson(DATA_STREAM_SERVICE_PATH)
+
+            val data = batch.let { mutableDataStreamBatch.appendBatch(it) }
+
+            verify {
+                mutableDataStreamBatch.appendBatch(batch)
+            }
+            println("Data after appending: $data")
+        }
+
+        @Test
+        fun `should throw if study deployment ID is not matched with dataStreamConfiguration's`() = runTest {
+
+            val studyDeploymentId = UUID.randomUUID()
+            val sut = CoreDataStreamService(
+                mockk(),
+                mockk(),
+                mockk(),
+                objectMapper
+            )
+
+            coEvery { dataStreamConfigurationRepository.findById(studyDeploymentId.stringRepresentation) } returns any()
+
+            val batch = readBatchFromJson(DATA_STREAM_SERVICE_PATH)
+
+            assertFailsWith<IllegalArgumentException> {
+                sut.appendToDataStreams(studyDeploymentId, batch)
+            }
+
+            coVerify(exactly = 0) {
+                dataStreamConfigurationRepository.findById(studyDeploymentId.stringRepresentation)
+            }
+        }
+
+        @Test
+        fun `should throw if study deployment ID does not match with the batch's`() = runTest {
+
+            val studyDeploymentId = UUID.randomUUID()
+
+            val sut = CoreDataStreamService(
+                mockk(),
+                mockk(),
+                mockk(),
+                objectMapper
+            )
+
+            val dataStreamConfigurationRepository = mockk<DataStreamConfigurationRepository>()
```

Comment:
You are not providing this mock in the constructor of the SUT. So you are not actually testing what you say you are testing here. I assume you get an exception, but only because there is no `findById` available on the mock.

Reply:
yes, it passed because findById is not available, not I provided it to the mock and still passes but in this case since the generated UUID is random (incorrect) it fails with coVerify(exactly = 0) upon calling

assertFailsWith {
sut.appendToDataStreams(studyDeploymentId, batch)
}